### DISCUSSION TITLE
Anyone tried nuxt for SEO?

### DISCUSSION THREAD

**Discussion by JenuelDev** [nuxt/nuxt#16673] [Ideas]
I know nuxt has ability to add meta tags easily to manage the meta tags for SEO. Any recommendations or suggestions or any concept that you wanna share about this.

**Comment by danielroe** [comment]
https://v3.nuxtjs.org/docs/usage/meta-tags/

**Comment by kimael-code** [comment]
Hi, I created my portfolio with Nuxt 3 and deployed it to Github pages. After running `npx nuxi generate`, the generated `index.html` contains within the `<head>` section only link tags. The SEO meta tags are somehow injected after those link tags mentioned above, causing the SEO functionality not to work properly because the only tags that search engines will receive are link tags about preloaded scripts and css files . In Nuxt 2, all meta and SEO tags are generated at the beginning of the `<head>` section in `index.html` making SEO work properly; in Nuxt 3, this behavior changes: meta and SEO tags are placed at the end.

**Reply by kimael-code** [reply]
To fix this. I had to write in hardcoded way all SEO and meta tags in the `<head>` section of the `index.html` after running `npx nuxi generate`. Much appreciate it if someone tells me how to fix this, because I've followed all official docs about generating SEO and meta tags. All SEO and meta tags are generated correctly but are placed in wrong order inside the `<head>` section of the `index.html` when making an static site, I do not know if this behavior occurs when rendering dynamically.

**Comment by JenuelDev** [comment]
After reviewing Nuxt 3, I think `SEO` is lot better when it is in SSR mode, this way it renders page on page load, or change route/page. You can google important meta tags or you can check this article: https://www.searchenginejournal.com/important-tags-seo/156440/#close

**Reply by TheAlexLichter** [reply]
SEO is always better (given same content) when the info is given in the HTML already and Google doesn't have to render the page via JavaScript ☺️
